python/lsst/drp/tasks/gbdesAstrometricFit.py

```diff
@@ -1164,9 +1181,15 @@ def _add_objects(self, wcsf, inputCatalogRefs, sourceIndices, extensionInfo, col
             for detector in detectors:
                 detectorSources = inputCatalog[inputCatalog["detector"] == detector]
 
-                extensionIndex = np.flatnonzero(
+                findExtension = np.flatnonzero(
                     (extensionInfo.visit == visit) & (extensionInfo.detector == detector)
-                )[0]
+                )
+                if len(findExtension) == 0:
+                    # This extension does not have information necessary for
+                    # fit. Skip these detections.
```

Comment:
Same comment as above, with the aded question of why this is duplicated here?

Reply:
This is duplicated because the sources get loaded two times--once when the source association is done, and again when they are added to the model fitting object.